internal/backup_push_handler.go

```diff
@@ -134,31 +136,64 @@ func HandleBackupPush(archiveDirectory string, uploader *Uploader) {
 	}
 
 	timelineChanged := bundle.checkTimelineChanged(conn)
-	var currentBackupSentinelDto *BackupSentinelDto
 
-	if !timelineChanged {
-		currentBackupSentinelDto = &BackupSentinelDto{
-			BackupStartLSN:   &backupStartLSN,
-			IncrementFromLSN: previousBackupSentinelDto.BackupStartLSN,
-			PgVersion:        pgVersion,
-		}
-		if previousBackupSentinelDto.BackupStartLSN != nil {
-			currentBackupSentinelDto.IncrementFrom = &previousBackupName
-			if previousBackupSentinelDto.isIncremental() {
-				currentBackupSentinelDto.IncrementFullName = previousBackupSentinelDto.IncrementFullName
-			} else {
-				currentBackupSentinelDto.IncrementFullName = &previousBackupName
-			}
-			currentBackupSentinelDto.IncrementCount = &incrementCount
-		}
+	if timelineChanged {
+		uploader.finish() // TODO : is it needed here?
```

Comment:
yes we should try to make consistent backup. The only problem with this timelineChanged thin is backup is named improperly. It still can be restored, but only on new timeline.
I think we should also upload that json sentinel, just make it unreadable....